src/redux-middleware/pullQueue.ts

```diff
@@ -193,9 +193,10 @@ const pullQueueMiddleware: ThunkMiddleware<State> = ({ getState, dispatch }) =>
       pullQueue = { ...pullQueue, ...initialPullQueue() }
       updatePullQueue({ forcePull: true })
     }
+    // do not pull before cursor has been initialized
     // do not updatePullQueue if there are syncs queued or in progress
     // this gets checked again in updatePullQueue, but short circuit here if possible
-    else if (!hasPushes(getState())) {
+    else if (getState().cursorInitialized && !hasPushes(getState())) {
```

Comment:
Out of curiosity, what happens if this check is removed?

Reply:
I think we would want `pullQueue` to start pulling after the cursor has been initialized. Else it will start pulling from the root context right from the first action dispatched. It makes sense to me that `pullQueue` starts pulling only when a cursor is known.

Reply:
Yes, that makes sense. We don't want redundant pulls.

Note what would happen if `pullQueue` handled the initial pull. Multiple pulls would automatically be deduped with the `lastExpanded` comparison. This is good because instead of it being an edge case, it is handled as a normal case, which reduces complexity. 

In this case the two approaches are pretty close to me, so this doesn't necessitate a change. Just wanted to highlight this as an important architectural consideration.